alpine/PenningTrapManager.h

```diff
@@ -229,46 +225,52 @@ class PenningTrapManager
 
             Kokkos::fence();
 
-            this->loadbalancer_m->initializeORB(FL_m.get(), mesh_m.get());
-            this->loadbalancer_m->repartition(FL_m.get(), mesh_m.get(), this->isFirstRepartition);
+            loadbalancer_m->initializeORB(FL.get(), mesh.get());
+            loadbalancer_m->repartition(FL.get(), mesh.get(), isFirstRepartition_m);
+            IpplTimings::stopTimer(domainDecomposition);
         }
 
+	static IpplTimings::TimerRef particleCreation = IpplTimings::getTimer("particlesCreation");
+        IpplTimings::startTimer(particleCreation);
+
         // Sample particle positions:
         ippl::detail::RegionLayout<double, Dim, Mesh_t<Dim>> rlayout;
-        rlayout = ippl::detail::RegionLayout<double, Dim, Mesh_t<Dim>>(*FL_m, *mesh_m);
-        size_type totalP_m = this->totalP;
+        rlayout = ippl::detail::RegionLayout<double, Dim, Mesh_t<Dim>>(*FL, *mesh);
+        size_type totalP = totalP_m;
         int seed           = 42;
         using size_type    = ippl::detail::size_type;
         Kokkos::Random_XorShift64_Pool<> rand_pool64((size_type)(seed + 100 * ippl::Comm->rank()));
 
         using samplingR_t =
             ippl::random::InverseTransformSampling<double, Dim, Kokkos::DefaultExecutionSpace,
                                                    DistR_t>;
-        Vector_t<double, Dim> rmin_m = rmin;
-        Vector_t<double, Dim> rmax_m = rmax;
-        samplingR_t samplingR(distR, rmax_m, rmin_m, rlayout, totalP_m);
+        Vector_t<double, Dim> rmin = rmin_m;
+        Vector_t<double, Dim> rmax = rmax_m;
+        samplingR_t samplingR(distR, rmax, rmin, rlayout, totalP);
         size_type nlocal = samplingR.getLocalSamplesNum();
 
-        this->pcontainer_m->create(nlocal);
+        pcontainer_m->create(nlocal);
 
-        view_type* R_m = &this->pcontainer_m->R.getView();
-        samplingR.generate(*R_m, rand_pool64);
+        view_type* R = &(pcontainer_m->R.getView());
+        samplingR.generate(*R, rand_pool64);
 
-        view_type* P_m = &this->pcontainer_m->getP().getView();
+        view_type* P = &(pcontainer_m->P.getView());
 
         double muP[Dim] = {0.0, 0.0, 0.0};
         double sdP[Dim] = {1.0, 1.0, 1.0};
-        Kokkos::parallel_for(nlocal, ippl::random::randn<double, Dim>(*P_m, rand_pool64, muP, sdP));
+        Kokkos::parallel_for(nlocal, ippl::random::randn<double, Dim>(*P, rand_pool64, muP, sdP));
 
         Kokkos::fence();
         ippl::Comm->barrier();
 
-        this->pcontainer_m->getQ() = this->Q / this->totalP;
+        IpplTimings::stopTimer(particleCreation);
+
+        pcontainer_m->q = Q_m / totalP_m;
         m << "particles created and initial conditions assigned " << endl;
     }
 
     void advance() override {
-        if (this->step_method == "LeapFrog") {
+        if (stepMethod_m == "LeapFrog") {
```

Comment:
Should have an assert here for checking if the stepper is not at all specified or if anything other than `LeapFrog` is specified.

Reply:
Done here d36b610